### PR TITLE
[WIP] Keras - WandbClassificationCallback

### PR DESCRIPTION
**Pull request by MathisFederico**
Added WandbClassificationCallback for confusion matrix and confusion examples logging.

Not sure how to deal with imports and dependencies here.

Suggestions are welcome, tests needs to be done.

### CODE REVIEW

**Comment by MathisFederico** [comment]
Nice ! Keep me updated !
If you want some help with something, do not hesitate to ask !
I will be happy to work on that with you !

**Comment by MathisFederico** [comment]
> This is so cool!  Thanks for the PR.  I want to think about this a little more.  We can import the libraries on demand using `util.get_module` but I want to do this in a more general way.  I'm thinking of adding some interface methods to WandbCallback that let's you sub class it and define custom metrics, import libraries, etc. to enable more callbacks like this.  I'll take a closer look this weekend.

Any news? I'd be happy to help!

**Comment by MathisFederico** [comment]
Makes perfect sense!  Will do! Thanks!

**Comment by MathisFederico** [comment]
> I guess you could actually just import the modules in your custom **init** instead of making the `load_modules` hook, that's simpler. You'd still want to use the wandb.util.get_module trick to give users nice error messages.

Done some work, let me know what you think of it !

File: wandb/keras/__init__.py [hunk 1]
```diff
@@ -288,6 +288,16 @@ def _implements_test_batch_hooks(self):
 
     def _implements_predict_batch_hooks(self):
         return self.log_batch_frequency is not None
+    
+    def _custom_wandb_logs(self, when):
+        """ Return the dictionary of custom logs
```
[the remark this answers is not in the document]
Comment (MathisFederico) [reply wandb/keras/__init__.py:293]: Not sure how to add the note.
Will do for logs

File: wandb/keras/__init__.py [hunk 2]
```diff
@@ -613,4 +626,148 @@ def _save_model(self, epoch):
             self.save_model = False
 
 
-__all__ = ['WandbCallback']
+class WandbClassificationCallback(WandbCallback):
+
+    def __init__(self, log_confusion_matrix=False, confusion_examples=0, confusion_classes=5, **kwargs):
```
[the remark this answers is not in the document]
Comment (MathisFederico) [reply wandb/keras/__init__.py:631]: Will do

[the remark this answers is not in the document]
Comment (MathisFederico) [reply wandb/keras/__init__.py]: Done

File: wandb/keras/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MathisFederico) [reply wandb/keras/__init__.py:293]: Done

**Comment by MathisFederico** [comment]
> This is AWESOME! I've left a few *nits but I think we're close. We should also add a test, I'm happy to help on this one. For reference, this is a basic one: https://github.com/wandb/client/blob/master/tests/test_keras.py#L71
> 
> Specific test's can be run locally with `pytest tests/test_keras.py -k my_test'

It seem's that the test you referred to is failing (because of wandb_init_run fixture)

**Comment by vanpelt** [comment]
I just merged into a feature branch in the mainline, I'll fix the test and have a couple of our engineers review.

**Pull request merged**